api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -77,6 +77,17 @@
    */
   String TARGET_FILE_SIZE_BYTES = "target-file-size-bytes";
 
+  /**
+   * If the compaction should use the sequence number of the snapshot at compaction start time for new data files,
+   * instead of using the sequence number of the newly produced snapshot.
+   * <p>
+   * This avoids commit conflicts with updates that add newer equality deletes at a higher sequence number.
+   * <p>
+   * Defaults to true.
+   */
+  String USE_STARTING_SEQUENCE_NUMBER = "use-starting-sequence-number";
+  boolean USE_STARTING_SEQUENCE_NUMBER_DEFAULT = true;
```

Comment:
Now that this is true, do we have to ignore it with V1 Tables?

Reply:
Yeah I was thinking about that right now. Technically I think it has no harm for v1 tables, because the sequence number is always 0, and it is not read or written anywhere. Let me add a unit test for v1. Do you see any place this might affect v1 table?

Reply:
That's all I was thinking, V1 Tables don't have sequence numbers so I just wanted to make sure they don't break if we are trying to set them.